rfc/rfc-0000-allow-resources-to-report-status.md

```diff
@@ -0,0 +1,298 @@
+# Meta
+[meta]: #meta
+- Name: Allow Resources to Report Status
+- Start Date: 2022-03-22
+- Author(s): @emmjohnson, @martyspiewak
+- Status: Draft <!-- Acceptable values: Draft, Approved, On Hold, Superseded -->
+- RFC Pull Request: (leave blank)
+- Supersedes: (put "N/A" unless this replaces an existing RFC, then link to that RFC)
+
+# Summary
+[summary]: #summary
+
+The RFC proposes modifying the existing templates to allow template authors to indicate how Cartographer can read the state of the resource so that it can be reflected on the owner status.
+
+# Motivation
+[motivation]: #motivation
+
+- Why should we do this?
+  - Currently, a blueprint has no knowledge of the current status of the stamped resources. This RFC proposes how resources can report (and categorize) this information strictly for informational purposes.
+- What use cases does it support?
+  - This feature will enable developers to learn about the progression through the blueprint by looking at the owner status alone. It will also surface information about the health of the underlying resources directly on the owner.
```

Comment:
This certainly surfaces information about the health of the underlying resources directly on the owner.

I wonder if it will truly enable developers to learn about the progression through the blueprint by looking at the owner status alone. Given 1 change (code commit) in flight at a time, that will certainly be the case. Given multiple changes in flight, it's not clear that it will be possible to reason about the status of one change vs another.

Reply:
The functionality proposed is still resource tracing rather than artifact tracing. The health condition surfaced is the latest value reported by the stamped resource, this is identical to how outputs are detected on the stamped resource and passed along the blueprint.

Health is rarely used to identify the path through a system, but more focused on identifying active failure points. One way that Cartographer is distinct from a pipeline is that even when a resource is unhealthy and not producing new outputs, other downstream resources are still available and able to continue processing (they just won't receive the upstream output updates).

Reply:
I think we're on the same page. To be clear, I'm not advocating for a larger scope here. Rather this suggestion:

```suggestion
  - This feature will surface information about the health of the underlying resources directly on the owner.
```